src/routes/imgurRoutes.js

```diff
@@ -0,0 +1,73 @@
+const userRoutes = require('./userRoutes');
+const express = require('express');
+const fileUploadMiddleware = require('express-fileupload');
+const { ImgurClient } = require('imgur');
+const User = require('../models/User');
+
+const router = express.Router();
+const imgurClient = new ImgurClient({
+  clientId: process.env.IMGUR_CLIENT_ID,
+  clientSecret: process.env.IMGUR_CLIENT_SECRET,
+  refreshToken: process.env.IMGUR_REFRESH_TOKEN,
+});
+
+userRoutes.post(
+  '/:userId/uploadProfile',
+  fileUploadMiddleware({ limits: { fileSize: 10000000 }, abortOnLimit: true }),
+  async (req, res) => {
+    console.log('Received request for /api/v1/profile/:userId/uploadDB');
+    const userId = req.params.userId;
+    console.log('User ID:', userId);
+    console.log('Entire req object:', req);
+    if (!userId) {
+      return res.status(400).send('No user id was provided.');
+    }
+
+    if (!req.files || Object.keys(req.files).length === 0) {
+      console.log('No files were uploaded.');
+      return res.status(400).send('No files were uploaded.');
+    }
+    if (!req.params.userId) {
+      return res.status(400).send('No user id was provided.');
+    }
+
+    const user = await User.findById(userId);
+    console.log('User:', user);
+    if (!user) {
+      return res.status(400).send('no user found');
+    }
+    const profile = req.files.profile;
+    console.log('File Details:', profile);
+
+    if (!profile) {
+      console.log('No profile file in the request.');
+      return res.status(400).send('No profile file in the request.');
+    }
+    console.log('File MIME Type:', profile.mimetype);
+    console.log('File Size:', profile.size);
+
+    if (!profile || !/^image/.test(profile.mimetype)) {
+      return res.status(400).send('File is not an image.');
+    }
+    if (!req.files || !req.files.profile) {
+      console.log('No profile file in the request.');
+      return res.status(400).send('No profile file in the request.');
+    }
+
+    const imgurRes = await imgurClient.upload({
+      image: profile.data.toString('base64'),
+      type: 'base64',
+      title: `${user._id}_profile_pic`,
+    });
+    if (imgurRes.success) {
+      user.profileImage = {
+        url: imgurRes.data.link,
+        altText: 'user profile picture',
+      };
+    }
+    res.status(200).send('Updated profile picture');
+    res.render('../views/imgur', { userId: req.params.userId });
```

Comment:
This is still trying to return 2 responses; one plain text response and one HTML/EJS response; which won't work per previous comment

Reply:
I have deleted the plain text response , commit ba4adec530fe34b6d22e428f759d95c8376d4fb3